compiler/rustc_ast/src/ast.rs

```diff
@@ -1799,19 +1799,21 @@ impl UintTy {
 /// A constraint on an associated type (e.g., `A = Bar` in `Foo<A = Bar>` or
 /// `A: TraitA + TraitB` in `Foo<A: TraitA + TraitB>`).
 #[derive(Clone, Encodable, Decodable, Debug)]
-pub struct AssocTyConstraint {
+pub struct AssocConstraint {
     pub id: NodeId,
     pub ident: Ident,
     pub gen_args: Option<GenericArgs>,
-    pub kind: AssocTyConstraintKind,
+    pub kind: AssocConstraintKind,
     pub span: Span,
 }
 
-/// The kinds of an `AssocTyConstraint`.
+/// The kinds of an `AssocConstraint`.
 #[derive(Clone, Encodable, Decodable, Debug)]
-pub enum AssocTyConstraintKind {
-    /// E.g., `A = Bar` in `Foo<A = Bar>`.
+pub enum AssocConstraintKind {
+    /// E.g., `A = Bar` in `Foo<A = Bar>` where A is an associated type.
     Equality { ty: P<Ty> },
```

Comment:
Seems like this variant should probably be renamed as well given the name change of the enum, perhaps this?

```suggestion
    TyEquality { ty: P<Ty> },
```

Reply:
Will handle this at some point if I maintain the additional variant!